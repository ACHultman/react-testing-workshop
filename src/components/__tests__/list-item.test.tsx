import { cleanup, render } from "@testing-library/react";
import ListItem from "../list-item";

describe("ListItem component", () => {
  afterEach(() => {
    cleanup();
  });

  it("matches snapshot", () => {
    const { baseElement } = render(<ListItem text={""} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("renders proper text", () => {
    const mockText = "test";
    const { getByText } = render(<ListItem text={mockText} />);
    getByText(mockText);
  });
});
