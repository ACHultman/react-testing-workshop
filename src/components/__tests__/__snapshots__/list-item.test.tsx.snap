// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component matches snapshot 1`] = `
<body>
  <div>
    <li
      class="list-item"
      data-testid="list-item"
    />
  </div>
</body>
`;
